rest-api/src/main/java/gov/cms/qpp/conversion/api/services/AuditServiceImpl.java

```diff
@@ -157,7 +160,8 @@ private Metadata initMetadata(Converter.ConversionReport report, MetadataHelper.
 	 */
 	private CompletableFuture<String> storeContent(Source sourceToStore) {
 		UUID key = UUID.randomUUID();
-		return storageService.store(key.toString(), sourceToStore.toInputStream(), sourceToStore.getSize());
+		Supplier<InputStream> stream = MeasuredInputStreamSupplier.terminallyTransformInputStream(sourceToStore.toInputStream());
```

Comment:
The `Source` parameter could already be an `InputStreamSupplierSource`.  So, by calling `MeasuredInputStreamSupplier#terminallyTransformInputStream`, we make another copy of the original `InputStream`.  What if we were to go into `Converter.ConversionReport` and change it to create `MeasuredInputStreamSupplier`s instead of `Source`s?  That way we don't have to make unnecessary copies of the data or take the time to make the copy.

Reply:
The source in Converter.ConversionReport is inherited from Converter -- no* extra work would be done to use a Source vs a MeasuredInputStreamSupplier - Source would (most of the time) use a MeasuredInputStreamSupplier under the covers anyways.

Here's a different idea -- let me know what you think. We change the method toInputStream in Source to return a Supplier<InputStream>.

Reply:
Sure, we could change the method to return a `Supplier<InputStream>`.
As you stated, most of the time a `Source` has a `MeasuredInputStreamSupplier` under the covers, and calling `terminallyTransformInputStream` on that kind of `Source` will result in a duplicate byte array being allocated.  That's all I want to avoid.